jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1496,15 +1530,13 @@ public static enum State {
  * @see State
  */
 public State getState() {
-	synchronized(lock) {
+	if (started) {
 		if (threadRef == NO_REF) {
-			if (isDead()) {
-				return State.TERMINATED;
-			}
-			return State.NEW;
+			return State.TERMINATED;
```

Comment:
Can you provide clarification on why there is no synchronized block here, but there is one in the Thread.java from https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/927?

Reply:
Thread state is essentially invalid as soon as you query it, so we may be able to avoid synchronization here for performance.

Reply:
In pre JDK19, synchronization here is not needed as the `threadRef` field is guaranteed to be valid as long as thread is started, once it is set to NO_REF, it wont change again. But with JDK19+, there are cases where a thread will not have the `Thread.holder` field set while running (ie. Attach API threads and etc.) so the sync block is to prevent the `eetop` field being updated while the check is in progress.

Reply:
is it possible to define the `holder` field for all cases (before `started` is set to `true`) so we can avoid the `synchronized` block in the extension repo PRs (e.g. https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/927)?

Reply:
I will have to check if any external code have access to the thread object before the holder field is initialized.
I've only seen internal access to holder during testing, so maybe if we ensure the started field is always set after constructor completes, then we can remove the sync block.

Reply:
[Personal build] passed, I will update all the extension changes to remove the sync block